pages/index.js

```diff
@@ -1,44 +1,36 @@
 import { Inter } from "@next/font/google";
-import {
-  useSession,
-  getSession,
-} from "next-auth/react";
 import Navigation from "./navigation/Navigation";
 import Layout from './components/layout';
-import { readUser } from "./api/user";
 import  LandingPage from "./landingpage.js";
-
+import { useUser } from '@auth0/nextjs-auth0/client';
 
 const inter = Inter({ subsets: ["latin"] });
 
 export default function Home(props) {
-  const { data: session } = useSession();
+  const { user, isLoading, error } = useUser();
+
+  if (isLoading) return <div>Loading...</div>;
+  if (error) return <div>{error.message}</div>;
   
+  let formattedUser = null;
+  if (user) {
+    // Format user for naviagtion
+    const userMetadata = user['https://vapartners.org/app_metadata'];
+    formattedUser = {
+      email: user.email,
+      name: user.name,
+      admin: (userMetadata.va_partners.admin !== undefined) ? userMetadata.va_partners.admin : false,
+      hospitalRole: userMetadata.va_partners.hospitalRole
+    };
+  }
+
   return (
     <Layout>
-      <Navigation user={props.user} />
-      {session && !props.user && (
+      <Navigation user={formattedUser} />
+      {!formattedUser || (!formattedUser.admin && formattedUser.hospitalRole.length == 0) && (
```

Comment:
Whole condition is very confusing and not making sense ...
if !formattedUser == true 
checking for !formattedUser.admin or formattedUser may cause nop. Better to change it to check the presence of  formattedUser.

Reply:
The problem is that even if a user is present, they still may not have any roles assigned to them. Which is why we want to display this warning if thats the case. If they aren't an Admin, and have no roles, then they should see this message.

I can clean up the conditional to make it easier to understand, but I think the logic is right.